client/app/bundles/course/assessment/submission/reducers/codaveriFeedbackStatus.js

```diff
@@ -8,18 +10,33 @@ export default function (state = initialState, action) {
   switch (action.type) {
     case actions.FETCH_SUBMISSION_SUCCESS:
     case actions.FINALISE_SUCCESS: {
-      return {
-        ...state,
-        answers: {
-          ...action.payload.answers.reduce(
-            (obj, answer) => ({
-              ...obj,
-              [answer.fields.id]: answer.codaveriFeedback,
-            }),
-            {},
-          ),
-        },
-      };
+      return produce(state, (draft) => {
+        draft.answers = action.payload.answers.reduce(
+          (reducerObject, answer) => {
+            reducerObject[answer.fields.id] = answer.codaveriFeedback;
+            return reducerObject;
+          },
+          {},
+        );
+      });
+    }
+    case actions.FEEDBACK_REQUEST: {
+      const { answerId } = action;
+      return produce(state, (draft) => {
+        draft.answers[answerId] = {
```

Comment:
Any difference with `draft.answers[answerId].jobStatus = 'errored';`?

If it's all the same, this should be more efficient since there's no need to recreate the object. We are already using immer anyway.

Reply:
There could be a case when `draft.answers[answerId]` doesnt exist yet.